### PR TITLE
GET | cars with user info

### PR DESCRIPTION
**Pull request by abdou059**

### CODE REVIEW

File: server/src/routes/cars.ts [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+} from '../controllers';
 import authMiddleware from '../middlewares/authMiddleware';
 
 const carsRouter = Router();
 
 carsRouter.get('/', ExpressWrapper(getFilteredCars));
 carsRouter.get('/', ExpressWrapper(getCarsById));
-carsRouter.put('/:id', authMiddleware('admin'), ExpressWrapper(updateCar));
+carsRouter.put('/:id', authMiddleware('admin'), ExpressWrapper(updateCars));
+carsRouter.get('/admin/cars', ExpressWrapper(getCarsDetails));
```
Comment (LinaYahya) [inline server/src/routes/cars.ts]:
```suggestion
carsRouter.get('/admin', authMiddleware('admin'), ExpressWrapper(getCarsDetails));
```

File: server/src/controllers/cars/getCarsDetails.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Request } from 'express';
+import { getCarsCustomInfo } from '../../services';
+
+const getCarsDetails = async (req: Request) => {
+  const {
+    state = '',
+    page = 1,
+  } = req.query;
```
Comment (LinaYahya) [inline server/src/controllers/cars/getCarsDetails.ts:9]: state here is required, So check before

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by mai-mohammed — APPROVED** [review]

File: server/src/controllers/cars/getFilteredCars.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ const getFilteredCars = async (req: Request) => {
   });
 
   if (result.rows.length === 0) {
+    // console.log(result.rows);
```
Comment (LinaYahya) [inline server/src/controllers/cars/getFilteredCars.ts]:
```suggestion
```

File: server/src/controllers/cars/getCarsDetails.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  if (!state) {
+    return { status: 200, msg: 'Not found' };
```
Comment (LinaYahya) [inline server/src/controllers/cars/getCarsDetails.ts]:
```suggestion
    throw createErr(400, 'your msg goes here');
```

File: server/src/services/cars/getCarsCustomInfo.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { Car, Customer } from '../../db/models';
+
+const CAR_NUM_IN_PAGE = 10;
+const getCarsDetailsQuery = async (state, page) => {
```
Comment (LinaYahya) [inline server/src/services/cars/getCarsCustomInfo.ts]:
```suggestion
const getCarsWithUserInfo = async (state, page) => {
```

File: server/src/routes/cars.ts [hunk 6]
```diff
@@ -2,13 +2,14 @@ import { Router } from 'express';
...
 import {
-  getFilteredCars, getCarsById, updateCar, deleteCarsById,
+  getFilteredCars, getCarsById, updateCar, deleteCarsById, getCarsDetails,
 } from '../controllers';
 
 const carsRouter = Router();
 
 carsRouter.get('/', ExpressWrapper(getFilteredCars));
 carsRouter.get('/', ExpressWrapper(getCarsById));
+carsRouter.get('/admin/cars', authMiddleware('admin'), ExpressWrapper(getCarsDetails));
```
Comment (LinaYahya) [inline server/src/routes/cars.ts]:
```suggestion
carsRouter.get('/admin', authMiddleware('admin'), ExpressWrapper(getCarsDetails));
```

File: server/src/services/cars/getCarsCustomInfo.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  const cars = await Car.findAndCountAll({
+    where: {
+      state,
+    },
+    offset: (page - 1) * CAR_NUM_IN_PAGE,
+    include: [
+      { model: Customer, attributes: ['fullName', 'phoneNumber', 'email'] },
+    ],
+    limit: CAR_NUM_IN_PAGE,
```
Comment (LinaYahya) [inline server/src/services/cars/getCarsCustomInfo.ts]:
```suggestion
    limit: CAR_NUM_IN_PAGE,
    distinct: true
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by husamkamal — APPROVED** [review]

**Pull request merged**